File: deaf_grandma.js

    var prompt = require('prompt'); //npm install prompt package at "https://www.npmjs.com/package/prompt". require('prompt') allows you to access all the tools associated with prompt

    prompt.start(); //start the prompt
    console.log("HEY KID!");
    let GOODBYE=0;
    
    const grandma = () => {
        prompt.get(["answer"], function (err, result) { //result.answer will allow you to capture the input from the prompt
          if  (result.answer === "") {
            console.log("WHAT?!");
            grandma() //this re-runs the function to allow you to continue running remaining conditionals 
          } else if (result.answer === result.answer.toLowerCase()) {
            console.log("SPEAK UP, KID!");
            grandma()
          } else if (result.answer === result.answer.toUpperCase() && result.answer !== "GOODBYE!") {
            console.log("NO, NOT SINCE 1946!");
            grandma()
        } else if (result.answer === "GOODBYE!" && GOODBYE === 1) {
            console.log("LATER SKATER!"); //Have to place this before the first goodbye conditional because we need to run the following conditional to add +1 to the "GOODBYE" variable. 
            return; //ends the function per instructions
          } else if (result.answer === "GOODBYE!") {
            GOODBYE++; 
            console.log("LEAVING SO SOON?");
            grandma()
          }
        });
    }
    grandma(); //invokes the function 

File: deaf_grandma.py

from typing import Counter


print("HEY KID!")
def deaf_grandma(): 
   a = input("Type Response Here: ")
   count = 0 #count is initialized outside the loop because it would drop back to 0 everytime the loop is ran 
   while(a): #while there is input continue looping 
    if a == "":
        print("WHAT?!")    
    elif a == a.lower():
        print("SPEAK UP, KID")   

    elif a == a.upper() and a != "GOODBYE!":
        print("NO, NOT SINCE 1946!")  

    elif a == "GOODBYE!" and count >= 1: #the second goodbye needs to be above the first goodbye, because we need it to register the count as 1 to prompt ("Later Skater"). The previous loop will add 1 to c.
        print("LATER, SKATER!")
        break
    
    elif a == "GOODBYE!":
        count += 1
        print("LEAVING SO SOON?")
          
    a = input("Type Response Here: ") #required to have this at the end of the loop to enable us to continue with more responses
    #print(c)

deaf_grandma()

